src/Model/helpers.js

```diff
@@ -23,22 +24,18 @@ export const createTimestampsFor = (model: Model) => {
 }
 
 export async function fetchChildren(model: Model): Promise<Model[]> {
-  const { associations } = model.collection.modelClass
-  const childrenKeys = Object.keys(associations).filter(key => associations[key].type === 'has_many')
-  
-  const promises = childrenKeys.map(async key => {
-    const children = await model[key].fetch()
-    const childrenPromises = children.map(async child => {
-      return fetchChildren(child)
-    })
-    const grandchildren = await Promise.all(childrenPromises)
-    let result = []
-    grandchildren.forEach(elt => {result = [...result, ...elt]})
-    return [...result, ...children]
+  const hasManyAssociations = Object.entries(model.constructor.associations).filter(([, value]) => value.type === 'has_many')
+  const childrenQueries = hasManyAssociations.map(([key, value]) => {
+    const childCollection = model.collections.get(key)
+    return childCollection.query(Q.where(value.foreignKey, model.id))
   })
 
-  const results = await Promise.all(promises)
-  let descendants = []
-  results.forEach(res => {descendants = [...descendants, ...res]})
-  return descendants
+  const childPromise = async query => {
+    const children = await query.fetch()
+    const grandchildren = await allPromises(child => fetchChildren(child), children)
+    return Array.prototype.concat.apply([], grandchildren).concat(children)
```

Comment:
you can use `unnest` from `rambdax` to flatten the array, `Array.prototype.concat.apply([],` - that's just ugly

Reply:
done